From 7b23d89f0f666385dce66f56f2500c4280c5d7bf Mon Sep 17 00:00:00 2001
From: dianlujitao <dianlujitao@lineageos.org>
Date: Mon, 12 Nov 2018 19:49:05 +0800
Subject: [PATCH 30/46] Binder: Fix improper JNI call for dumpProxyDebugInfo

 * dumpProxyDebugInfo doesn't have a return value, so it should be
   invoked via CallStaticVoidMethod instead of CallStaticObjectMethod.

Change-Id: Ie4c497f69b371ec3d55e7736827f73142d14a5b4
---
 core/jni/android_util_Binder.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/jni/android_util_Binder.cpp b/core/jni/android_util_Binder.cpp
index c3ba9ba8282..76ea4ed699b 100644
--- a/core/jni/android_util_Binder.cpp
+++ b/core/jni/android_util_Binder.cpp
@@ -1010,8 +1010,8 @@ static void android_os_BinderInternal_proxyLimitcallback(int uid)
     {
         // Calls into BinderProxy must be serialized
         AutoMutex _l(gProxyLock);
-        env->CallStaticObjectMethod(gBinderProxyOffsets.mClass,
-                                    gBinderProxyOffsets.mDumpProxyDebugInfo);
+        env->CallStaticVoidMethod(gBinderProxyOffsets.mClass,
+                                  gBinderProxyOffsets.mDumpProxyDebugInfo);
     }
     if (env->ExceptionCheck()) {
         ScopedLocalRef<jthrowable> excep(env, env->ExceptionOccurred());
-- 
2.17.1

